recipes/curlpp/all/conanfile.py

```diff
@@ -50,26 +40,23 @@ def requirements(self):
         self.requires("libcurl/8.9.1", transitive_headers=True, transitive_libs=True)
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            check_min_cppstd(self, self._min_cppstd)
-        else:
-            if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "6":
-                raise ConanInvalidConfiguration("${self.ref} requires C++11. Please set 'compiler.cppstd=11'.")
-            if self.settings.compiler == "apple-clang" and Version(self.settings.compiler.version) < "14":
-                raise ConanInvalidConfiguration("${self.ref} requires C++11. Please set 'compiler.cppstd=11'.")
+        check_min_cppstd(self, 11)
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "6":
+            raise ConanInvalidConfiguration("${self.ref} requires C++11. Please set 'compiler.cppstd=11'.")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
 
     def generate(self):
         tc = CMakeToolchain(self)
-        tc.variables["CURLPP_BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["CURLPP_BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["CMAKE_POLICY_VERSION_MINIMUM"] = "3.5"  # CMake 4 support
```

Comment:
Maybe we should add this check, equivalent to other CMake 4 PRs, as in the future this library could add support, being this `cache_variable` definition incorrect. 
(If added, don't forget importing proper `ConanException`)

```suggestion
        tc.cache_variables["CMAKE_POLICY_VERSION_MINIMUM"] = "3.5"  # CMake 4 support
        if Version(self.version) > "0.8.1.cci.20240530": # pylint: disable=conan-unreachable-upper-version
             raise ConanException("CMAKE_POLICY_VERSION_MINIMUM hardcoded to 3.5, check if new version supports CMake 4")

```